nmdc_server/cli.py

```diff
@@ -3,54 +3,37 @@
 from typing import Optional
 
 import click
+from alembic import command
+from alembic.config import Config
 
-from nmdc_server import jobs, models
-from nmdc_server.config import Settings, settings
-from nmdc_server.database import SessionLocal, SessionLocalIngest
+from nmdc_server import database
+from nmdc_server.config import settings
 from nmdc_server.ingest import errors
 from nmdc_server.ingest.all import load
-from nmdc_server.ingest.common import maybe_merge_download_artifact
 from nmdc_server.logger import get_logger
 
 
 @click.group()
 @click.pass_context
 def cli(ctx):
-    settings = Settings()
     if settings.environment == "testing":
         settings.database_uri = settings.testing_database_uri
     ctx.obj = {"settings": settings}
 
 
 @cli.command()
-@click.option("--ingest-db", is_flag=True, default=False)
-def migrate(ingest_db: bool):
+def migrate():
     """Upgrade the database schema."""
-    jobs.migrate(ingest_db=ingest_db)
+    database_uri = settings.current_db_uri
+    session_maker = database.SessionLocal
 
-
-@cli.command()
-def truncate():
-    """Remove all existing data from the ingest database."""
-    with SessionLocalIngest() as db:
-        try:
-            db.execute("select truncate_tables()").all()
-            db.commit()
-        except Exception:
-            db.rollback()
-            db.execute(
-                """
-                DO $$ DECLARE
-                     r RECORD;
-                 BEGIN
-                     FOR r IN (SELECT tablename FROM pg_tables WHERE schemaname = current_schema())
-                     LOOP
-                         EXECUTE 'DROP TABLE IF EXISTS ' || quote_ident(r.tablename) || ' CASCADE';
-                     END LOOP;
-                 END $$;
-            """
-            )
-            db.commit()
+    with session_maker.begin() as db:  # type: ignore
+        database.get_ingest_lock(db)
+        alembic_cfg = Config(str(Path(__file__).parent / "alembic.ini"))
+        alembic_cfg.set_main_option("script_location", str(Path(__file__).parent / "migrations"))
+        alembic_cfg.set_main_option("sqlalchemy.url", database_uri)
+        alembic_cfg.attributes["configure_logger"] = True
+        command.upgrade(alembic_cfg, "head")
```

Comment:
This invariably runs the migration which results in a no-op if no migrations are required. However, the migration command must acquire the [ingest lock](https://github.com/microbiomedata/nmdc-server/blob/8a97b502a278f40c6d7b0820e58768f1fd223702/nmdc_server/database.py#L175-L181).

Since our [container invokes the `nmdc-server migrate` command on startup](https://github.com/microbiomedata/nmdc-server/blob/8a97b502a278f40c6d7b0820e58768f1fd223702/prestart.sh#L15), a currently running ingestion will block the container from even starting as the migrate command will throw an error if the lock cannot be acquired. 

This is desirable behavior if migrations must be applied, but bad if the migration command would be a no-op. We need to change this to only conditionally apply migrations if required. [Luckily, this has been implemented in the past.](https://github.com/microbiomedata/nmdc-server/blob/42ed6fce223cafe60b618918fb2e9921131d4107/nmdc_server/jobs.py#L44-L51)